news-explorer-frontend/src/components/PopupWithForm/PopupWithForm.js

```diff
@@ -0,0 +1,310 @@
+import React, { useEffect, useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import closebtn from '../../images/close.svg';
+import { authorize, register } from '../../utils/auth';
+import './PopupWithForm.css';
+
+function PopupWithForm(props) {
+  const isSignInPopupOpen = props.isSignInPopupOpen;
```

Comment:
`PopupWithForm` should be a universal component. 

According to the task you need to have 2 different popups with forms: `Login` and `Register` . These 2 components shoulld use `PopupWithForm` inside them, passing their inputs, submit handlers and so on to the `props` . The inputs as `chidren`, other info as `props`